presto-main/src/main/java/com/facebook/presto/execution/DropFunctionTask.java

```diff
@@ -53,19 +58,29 @@ public String getName()
     @Override
     public String explain(DropFunction statement, List<Expression> parameters)
     {
-        return "DROP FUNCTION " + statement.getFunctionName();
+        return format("DROP %sFUNCTION %s", statement.isTemporary() ? "TEMPORARY " : "", statement.getFunctionName());
     }
 
     @Override
     public ListenableFuture<?> execute(DropFunction statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
     {
         Analyzer analyzer = new Analyzer(stateMachine.getSession(), metadata, sqlParser, accessControl, Optional.empty(), parameters, stateMachine.getWarningCollector());
         analyzer.analyze(statement);
+        Optional<List<TypeSignature>> parameterTypes = statement.getParameterTypes().map(types -> types.stream().map(TypeSignature::parseTypeSignature).collect(toImmutableList()));
 
-        metadata.getFunctionManager().dropFunction(
-                qualifyFunctionName(statement.getFunctionName()),
-                statement.getParameterTypes().map(types -> types.stream().map(TypeSignature::parseTypeSignature).collect(toImmutableList())),
-                statement.isExists());
+        if (statement.isTemporary()) {
+            stateMachine.removeSessionFunction(
+                    new BasicSignature(
+                            QualifiedFunctionName.of(SESSION_NAMESPACE, statement.getFunctionName().getSuffix()),
+                            parameterTypes.orElse(emptyList())),
+                    statement.isExists());
```

Comment:
Where is the error if the function doesn't exist?

Reply:
QueryStateMachine will throw an exception if the function doesn't exist, in method "removeSessionFunction"